src/main/java/com/microsoft/sqlserver/jdbc/SQLCollation.java

```diff
@@ -602,7 +600,23 @@ final Encoding checkSupported() throws UnsupportedEncodingException
         return this;
     }
 
-    final String charsetName() { return charsetName; }
+    final Charset charset() throws SQLServerException
+    {
+        try
+        {
+            checkSupported();
+            if (charset == null)
+            {
+                charset = Charset.forName(charsetName);
```

Comment:
@marschall thank you very much for your great idea and fix. I just have one more question and hope you could consider. Is it possible to delete the local variable 'private Charset charset'? I am not sure why we need to check if the variable is null here. instead of assigning it to charset and returning charset., could we just do "return Charset.forName(charsetName);" ?